src/Kovan-DssSpell.t.sol

```diff
@@ -542,58 +606,84 @@ contract DssSpellTest is DSTest, DSMath {
-        // Convert whole Dai units to expected RAD
-        uint normalizedTestDunk = values.collaterals[ilk].dunk * RAD;
-        assertEq(dunk, normalizedTestDunk);
-        // put back in after LIQ-1.2
-        assertTrue(dunk >= RAD && dunk < MILLION * RAD);
-        }
-        {
-        (,uint mat) = spot.ilks(ilk);
-        // Convert BP to system expected value
-        uint normalizedTestMat = (values.collaterals[ilk].mat * 10**23);
-        assertEq(mat, normalizedTestMat);
-        assertTrue(mat >= RAY && mat < 10 * RAY);    // cr eq 100% and lt 1000%
-        }
-        {
-        (address flipper,,) = cat.ilks(ilk);
-        FlipAbstract flip = FlipAbstract(flipper);
-        // Convert BP to system expected value
-        uint normalizedTestBeg = (values.collaterals[ilk].beg + 10000)  * 10**14;
-        assertEq(uint(flip.beg()), normalizedTestBeg);
-        assertTrue(flip.beg() >= WAD && flip.beg() < 105 * WAD / 100);  // gt eq 0% and lt 5%
-        assertEq(uint(flip.ttl()), values.collaterals[ilk].ttl);
-        assertTrue(flip.ttl() >= 600 && flip.ttl() < 10 hours);         // gt eq 10 minutes and lt 10 hours
-        assertEq(uint(flip.tau()), values.collaterals[ilk].tau);
-        assertTrue(flip.tau() >= 600 && flip.tau() <= 1 hours);          // gt eq 10 minutes and lt eq 1 hours
-
-        assertEq(flip.wards(address(cat)), values.collaterals[ilk].liquidations);  // liquidations == 1 => on
+    function checkCollateralValues(SystemValues storage values) internal {
+        uint256 sumlines;
+        bytes32[] memory ilks = reg.list();
+        for(uint256 i = 0; i < ilks.length; i++) {
+            bytes32 ilk = ilks[i];
+            (uint256 duty,)  = jug.ilks(ilk);
+
+            assertEq(duty, rates.rates(values.collaterals[ilk].pct));
+            // make sure duty is less than 1000% APR
+            // bc -l <<< 'scale=27; e( l(10.00)/(60 * 60 * 24 * 365) )'
+            // 1000000073014496989316680335
+            assertTrue(duty >= RAY && duty < 1000000073014496989316680335);  // gt 0 and lt 1000%
+            assertTrue(diffCalc(expectedRate(values.collaterals[ilk].pct), yearlyYield(rates.rates(values.collaterals[ilk].pct))) <= TOLERANCE);
+            assertTrue(values.collaterals[ilk].pct < THOUSAND * THOUSAND);   // check value lt 1000%
+            {
+            (,,, uint256 line, uint256 dust) = vat.ilks(ilk);
+            // Convert whole Dai units to expected RAD
+            uint256 normalizedTestLine = values.collaterals[ilk].line * RAD;
+            sumlines += values.collaterals[ilk].line;
+            (uint256 aL_line, uint256 aL_gap, uint256 aL_ttl,,) = autoLine.ilks(ilk);
+            if (!values.collaterals[ilk].aL_enabled) {
+                assertTrue(aL_line == 0);
+                assertEq(line, normalizedTestLine);
+                assertTrue((line >= RAD && line < BILLION * RAD) || line == 0);  // eq 0 or gt eq 1 RAD and lt 1B
```

Comment:
Seeing several assertions of this form--might be worth defining a dedicated assertion function:
```
assertInRangeOrZero(uint256 val, uint256 min, uint256 max);
```

Reply:
We should do this assertion outside of the if block actually becuase it should apply to both conditions.